plugins/kubernetes-backend/src/service/KubernetesBuilder.ts

```diff
@@ -67,6 +79,37 @@ export type KubernetesBuilderReturn = Promise<{
   serviceLocator: KubernetesServiceLocator;
 }>;
 
+const RESOURCE_TYPE_KUBERNETES_RESOURCE = 'kubernetes-resource';
+type K8sQuery = Set<{
+  apiVersion: string;
+  kind: string;
+  cluster?: string;
+  namespace?: string;
+  labelSelector?: string;
+  subResource?: string;
+}>;
+const createKubernetesPermissionRule = makeCreatePermissionRule<
+  object,
+  K8sQuery,
+  typeof RESOURCE_TYPE_KUBERNETES_RESOURCE
+>();
+const matchesJSON = createKubernetesPermissionRule({
+  name: 'MATCHES_JSON',
+  description: 'Allow kubernetes resources matching the given JSON structure',
+  resourceType: RESOURCE_TYPE_KUBERNETES_RESOURCE,
+  apply: (resource: object, params: PermissionRuleParams) => {
```

Comment:
If we expanded this type to ensure that every resource had a field identifying the cluster it lived on, maybe `apply` could always return `true` for clusters with client-side auth; this might signify that for client-side auth providers Backstage is not going to prevent access to anything, it leaves the authorization decisions entirely up to k8s in those cases